### PR TITLE
Switch to Alpine linux for container base

### PR DESCRIPTION
**Pull request by ikke-t**
I converted this to use Alpine linux container for base in the progress to port this to ARM architecture for Raspberry Pi 3. While doing so I tried to keep the size minimal. So I removed sudo and vim. I tried to get away without bash, but for some reason the java BuildTools.jar requires bash. I already modified start-minecrft.sh to work in with sh, but revered it to bash. Can we remove the requirement? Also I added git. It could be reduced, if we wget the source instead. Those two add ~250MB into image size, quite uselessly. Also perhaps imagemagic could be removed?

Anyhow, I tried this on my laptop, and Minecraft could connect to the server. Couple of exceptions appeared during java built, but perhaps they are not due these changes.

### CODE REVIEW

**Comment by itzg** [comment]
This update also works for me. Are good with me merging this?

Here is a size comparison of the current:
```
IMAGE               CREATED             CREATED BY                                      SIZE
0844a55c6138        3 months ago        /bin/sh -c #(nop)  ENV UID=1000 GID=1000 M...   0 B
<missing>           3 months ago        /bin/sh -c #(nop)  ENTRYPOINT ["/start"]        0 B
<missing>           3 months ago        /bin/sh -c #(nop)  WORKDIR /data                0 B
<missing>           3 months ago        /bin/sh -c #(nop) COPY file:14a558c3b10bdb...   715 B
<missing>           3 months ago        /bin/sh -c #(nop)  VOLUME [/data /mods /co...   0 B
<missing>           3 months ago        /bin/sh -c chmod +x /usr/local/bin/*            9.53 MB
<missing>           3 months ago        /bin/sh -c #(nop) COPY file:cedcf01950737c...   251 B
<missing>           3 months ago        /bin/sh -c #(nop) COPY file:4f68792f9bce36...   12 kB
<missing>           3 months ago        /bin/sh -c #(nop) COPY file:074c760870c29d...   377 B
<missing>           3 months ago        /bin/sh -c #(nop) ADD tarsum.v1+sha256:b78...   9.53 MB
<missing>           3 months ago        /bin/sh -c #(nop)  EXPOSE 25565/tcp 25575/tcp   0 B
<missing>           3 months ago        /bin/sh -c useradd -M -s /bin/false --uid ...   329 kB
<missing>           3 months ago        /bin/sh -c DEBIAN_FRONTEND=noninteractive ...   104 MB
<missing>           3 months ago        /bin/sh -c apt-get update                       10.8 MB
<missing>           3 months ago        /bin/sh -c #(nop)  ENV APT_GET_UPDATE=2016...   0 B
<missing>           3 months ago        /bin/sh -c #(nop)  MAINTAINER itzg              0 B
<missing>           4 months ago        /bin/sh -c /var/lib/dpkg/info/ca-certifica...   418 kB
<missing>           4 months ago        /bin/sh -c set -x  && apt-get update  && a...   349 MB
<missing>           4 months ago        /bin/sh -c #(nop) ENV CA_CERTIFICATES_JAVA...   0 B
<missing>           4 months ago        /bin/sh -c #(nop) ENV JAVA_DEBIAN_VERSION=...   0 B
<missing>           4 months ago        /bin/sh -c #(nop) ENV JAVA_VERSION=8u102        0 B
<missing>           4 months ago        /bin/sh -c #(nop) ENV JAVA_HOME=/usr/lib/j...   0 B
<missing>           4 months ago        /bin/sh -c {   echo '#!/bin/sh';   echo 's...   87 B
<missing>           4 months ago        /bin/sh -c #(nop) ENV LANG=C.UTF-8              0 B
<missing>           4 months ago        /bin/sh -c echo 'deb http://httpredir.debi...   61 B
<missing>           4 months ago        /bin/sh -c apt-get update && apt-get insta...   1.29 MB
<missing>           4 months ago        /bin/sh -c apt-get update && apt-get insta...   123 MB
<missing>           4 months ago        /bin/sh -c apt-get update && apt-get insta...   44.3 MB
<missing>           4 months ago        /bin/sh -c #(nop) CMD ["/bin/bash"]             0 B
<missing>           4 months ago        /bin/sh -c #(nop) ADD file:0e0565652aa852f...   125 MB
```

...and now with Alpine
```
IMAGE               CREATED             CREATED BY                                      SIZE
74614c51aa04        21 minutes ago      /bin/sh -c #(nop)  ENV UID=1000 GID=1000 M...   0 B
5ad69dae0ae7        21 minutes ago      /bin/sh -c #(nop)  ENTRYPOINT ["/start"]        0 B
f93e8ce0fce2        21 minutes ago      /bin/sh -c #(nop) WORKDIR /data                 0 B
200797610616        21 minutes ago      /bin/sh -c #(nop) COPY file:4020448e5818fb...   753 B
ac0a38e03496        21 minutes ago      /bin/sh -c #(nop)  VOLUME [/data /mods /co...   0 B
9b191bd88772        21 minutes ago      /bin/sh -c chmod +x /usr/local/bin/*            9.53 MB
98202dbe0a56        21 minutes ago      /bin/sh -c #(nop) COPY file:2ae2203ab1f775...   258 B
5aa080d34221        21 minutes ago      /bin/sh -c #(nop) COPY file:be2b370f51478c...   12.2 kB
0cef3ac9e7c4        21 minutes ago      /bin/sh -c #(nop) COPY file:c6e7f5c3641ef3...   495 B
7b66c2b0f77d        22 minutes ago      /bin/sh -c #(nop) ADD tarsum.v1+sha256:b78...   9.53 MB
4ad750db7a69        22 minutes ago      /bin/sh -c #(nop)  EXPOSE 25565/tcp 25575/tcp   0 B
2dca03dcbddd        22 minutes ago      /bin/sh -c addgroup -g 1000 minecraft   &&...   4.97 kB
a9a25df45ded        22 minutes ago      /bin/sh -c echo "http://dl-3.alpinelinux.o...   145 MB
bb829c2a1767        22 minutes ago      /bin/sh -c #(nop)  MAINTAINER itzg              0 B
4e38e38c8ce0        5 months ago        /bin/sh -c #(nop) ADD file:852e9d0cb9d9065...   4.8 MB
```

**Comment by itzg** [comment]
...or more concisely:

```
REPOSITORY              TAG                 IMAGE ID            CREATED             SIZE
itzg/minecraft-server   latest              0844a55c6138        3 months ago        777 MB

REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
mc                  latest              74614c51aa04        23 minutes ago      169 MB
```

**Comment by ikke-t** [comment]
Quite a nice change in size! I'm done with changes for now, go for it from my behalf.

**Comment by itzg** [comment]
Hmm, I need to figure out how to statically build https://github.com/itzg/restify since Alpine apparently does something quirky with libc vs musl...

**Comment by zaquestion** [comment]
@itzg 
```
CGO_ENABLED=0 GOOS=linux go build .
```

**Comment by itzg** [comment]
Ah, I'll give that a try.

File: minecraft-server/Dockerfile [hunk 1]
```diff
@@ -1,25 +1,26 @@
...
+        apk update && \
+        apk add \
+          openjdk8-jre-base \
+          openssl \
+          imagemagick \
+          lsof \
+          su-exec \
+          bash \
+          git \
+          jq &&\
```
Comment (itzg) [inline minecraft-server/Dockerfile]: Please also add `ca-certificates` so that x509 certs resolve while restify accesses https URLs.

**Comment by itzg** [comment]
Thanks @zaquestion , that little adjustment worked perfectly.

**Comment by itzg** [comment]
Sorry @ikke-t that it took me so long to resolve this PR.

Thanks @zaquestion for the pointer.

**Comment by itzg** [comment]
Pushed to hub with image ID `1bdc562b1d0d`

**Comment by ikke-t** [comment]
thanks, glad to see my exercise was helpful!

**Pull request merged**